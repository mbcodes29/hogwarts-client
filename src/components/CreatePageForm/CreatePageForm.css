.create-form {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    align-self: center;
    text-align: center;
    color: white;
    text-shadow:5px 5px 10px black;
    font-weight: bolder;
    margin: 0 5% 0 5%;
    font-weight: bolder; 
    background: rgba(0, 0, 0, .5);
}

.create-form:hover {
    cursor: url('../../images/cursor.png'), auto;
}

.select-class:hover {
    cursor: url('../../images/cursor.png'), auto;
}

.label-class:hover {
    cursor: url('../../images/cursor.png'), auto;
}

.option-class:hover {
    cursor: url('../../images/cursor.png'), auto;
}

.input-class:hover {
    cursor: url('../../images/cursor.png'), auto;
}