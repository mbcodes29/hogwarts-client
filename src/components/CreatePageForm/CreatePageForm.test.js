import React from 'react';
import ReactDOM from 'react-dom';
import CreatePageForm from './CreatePageForm'
import {BrowserRouter} from 'react-router-dom'
import renderer from "react-test-renderer"

it('renders without crashing',() => {
    const div = document.createElement('div')
    ReactDOM.render(
    <BrowserRouter>
        <CreatePageForm />
    </BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
})

it('renders the UI as expected', () => {
    const tree = renderer
      .create(<BrowserRouter><CreatePageForm /></BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });