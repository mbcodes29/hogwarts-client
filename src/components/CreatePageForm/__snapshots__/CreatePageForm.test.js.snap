// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<form
  className="create-form"
  onSubmit={[Function]}
>
  <label
    className="label-class"
  >
    Pick your student's preferred pronouns:
    <br />
    <select
      className="select-class"
      name="pronouns"
      onChange={[Function]}
      required={true}
    >
      <option
        className="option-class"
        value=""
      >
        select
      </option>
      <option
        value="she-her"
      >
        She/Her
      </option>
      <option
        value="he-him"
      >
        He/Him
      </option>
      <option
        value="they-them"
      >
        They/Them
      </option>
    </select>
  </label>
  <br />
  <br />
  <label
    className="label-class"
  >
    Choose a pet:
    <br />
    <select
      className="select-class"
      name="pet"
      onChange={[Function]}
      required={true}
    >
      <option
        className="option-class"
        value=""
      >
        select
      </option>
      <option
        value="Owl"
      >
        Owl
      </option>
      <option
        value="Cat"
      >
        Cat
      </option>
      <option
        value="Toad"
      >
        Toad
      </option>
    </select>
  </label>
  <br />
  <br />
  <label
    className="label-class"
  >
    Choose your wand type:
    <br />
    <select
      className="select-class"
      name="wandType"
      onChange={[Function]}
      required={true}
    >
      <option
        className="option-class"
        value=""
      >
        select
      </option>
      <option
        value="birch"
      >
        Birch
      </option>
      <option
        value="ash"
      >
        Ash
      </option>
      <option
        value="alder"
      >
        Alder
      </option>
      <option
        value="willow"
      >
        Willow
      </option>
      <option
        value="oak"
      >
        Oak
      </option>
      <option
        value="holly"
      >
        Holly
      </option>
      <option
        value="hazel"
      >
        Hazel
      </option>
      <option
        value="ivy"
      >
        Ivy
      </option>
    </select>
  </label>
  <br />
  <br />
  <label
    className="label-class"
  >
    Choose your wand core:
    <br />
    <select
      className="select-class"
      name="wandCore"
      onChange={[Function]}
      required={true}
    >
      <option
        className="option-class"
        value=""
      >
        select
      </option>
      <option
        value="unicorn"
      >
        Unicorn
      </option>
      <option
        value="dragon"
      >
        Dragon
      </option>
      <option
        value="phoenix"
      >
        Phoenix
      </option>
    </select>
  </label>
  <br />
  <br />
  <label
    className="label-class"
  >
    Choose a favorite subject:
    <br />
    <select
      className="select-class"
      name="favoriteSubject"
      onChange={[Function]}
      required={true}
    >
      <option
        className="option-class"
        value=""
      >
        select
      </option>
      <option
        value="transfiguration"
      >
        Transfiguration
      </option>
      <option
        value="charms"
      >
        Charms
      </option>
      <option
        value="potions"
      >
        Potions
      </option>
      <option
        value="history"
      >
        History of Magic
      </option>
      <option
        value="defense"
      >
        Defense Against the Dark Arts
      </option>
      <option
        value="astronomy"
      >
        Astronomy
      </option>
      <option
        value="herbology"
      >
        Herbology
      </option>
      <option
        value="flying"
      >
        Flying
      </option>
      <option
        value="divination"
      >
        Divination
      </option>
      <option
        value="magical-creatures"
      >
        Care of Magical Creatures
      </option>
    </select>
  </label>
  <br />
  <br />
  <input
    className="input-class"
    type="submit"
    value="Submit"
  />
</form>
`;
